src/lib.rs

```diff
@@ -124,8 +124,22 @@ pub fn init<P: AsRef<Path>>(paths: &[P], options: Options) -> std::io::Result<()
     let resources = ResourceManager::new();
     let base_dirs = dirs::ProjectDirs::from("io", "cloudhead", "rx")
         .ok_or_else(|| io::Error::new(io::ErrorKind::NotFound, "home directory not found"))?;
-    let mut session = Session::new(win_w, win_h, hidpi_factor, resources.clone(), base_dirs)
-        .init(options.source.clone())?;
+
+    let mut session = Session::new(win_w, win_h, hidpi_factor, resources.clone(), base_dirs);
+
+    // TODO: If this is always true move this to Session::new
+    if session.views.is_empty() {
```

Comment:
This will always be true if you move it here, so it's not required. What might be a problem though is that the blank view remains when you start the session with a bunch of files.

Reply:
We could use `blank` as a builder function - have it consume `self` and return `Self`, just like `init`.

Then we can do `let mut session = Session::new(....).blank(..).init(...);`!

Reply:
I think the behavior of destroying the scratch view when loading a new view already exists too, so you might not need to add anything there: https://github.com/cloudhead/rx/blob/59ac9cce2eb1b5e0f4606f21504e81c9b521ba87/src/session.rs#L1556-L1563

Reply:
Great feedback. I'll get to this tonight.

Related, do you think this warrants automated tests? Once I fix everything else that'll be the last change I can think to add.

Reply:
Could you clarify your last point? Do I need to do anything if I adopt your `Session::new(...).blank(...).init(...)?` pattern?

Reply:
@cloudhead Let me know what you think of my solution.
The [use of `Session::blank(...)` in `Session::edit(...)`](https://github.com/cloudhead/rx/blob/master/src/session.rs#L1439) made it really hard to refactor directly, but I was able to add a `Session::as_blank(...)` (name flexible of course) that chained nicely and didn't break existing use of `Sessoin::blank(...)`.
I'm not super happy with the workaround, but it gets the job done and I'm not experienced enough with Rust to know how to refactor `Session::blank(...)` directly without the need for `Session::as_blank(...)`.

Reply:
> Could you clarify your last point? Do I need to do anything if I adopt your `Session::new(...).blank(...).init(...)?` pattern?

You just have to change `Session::blank` a little bit to look more like `Session::init` - it should have a type signature like `fn blank(self) -> Self`. Then, `as_blank` won't be necessary.

Automated tests are not necessary.